src/dist_example.cpp

```diff
@@ -0,0 +1,223 @@
+#include <Omega_h_array_ops.hpp>
+#include <Omega_h_dist.hpp>
+#include <Omega_h_compare.hpp>
+#include <Omega_h_library.hpp>
+
+using namespace Omega_h;
+
+static void test_three_ranks(CommPtr comm){
+  OMEGA_H_CHECK(comm->size() == 3);
+  Dist dist;
+  dist.set_parent_comm(comm);
+/*
+ * Starting Partition: {0, 1}  {2, 3}  {4, 5}
+ * Reversal to       : {5, 4}  {3, 2}  {1, 0}
+*/
+  //Configure the directed graph by finding our destination node(reverse root)
+  int reverse_root = 2 - comm->rank();
+  dist.set_dest_ranks(Read<I32>({reverse_root, reverse_root}));
+/*
+ * Forward: Rank 0    Rank 1    Rank 2
+ *            │ ┏━━━━━━╾╫╼━━━━━━━┛                 
+ *            ╰─╂───────╫─────────╮
+ *            ┏━┛       ║         │
+ * Reverse: Rank 0    Rank 1    Rank 2
+*/ 
+  //Set the destination indexes of the reverse root
+  dist.set_dest_idxs(LOs({1, 0}), 2);
+
+  //Generate the Packets for the forward root(2 Packets of width 1)
+  int width = 1;
+  Reals a({2.0 * comm->rank(), 2.0 * comm->rank() + 1});
+
+  //Perform the exchange operation, and verify it was executed correctly
+  auto b = dist.exch(a, width);
+  OMEGA_H_CHECK(b == Reals({2.0 * reverse_root + 1, 2.0 * reverse_root}));
+}
+
+static void test_packet_fanout(CommPtr comm){
+  OMEGA_H_CHECK(comm->size() == 3);
+  Dist dist;
+  dist.set_parent_comm(comm);
+/*
+ * Starting Partition: {1, 2}  {0, 0}  {0, 0}
+ * End Partition:      {}      {1, 1}  {2, 2}
+*/
+
+  //Configure the directed graph. Note when explicitly setting parameters such as roots2items, 
+  //undefined behavior may occur when not set for every rank. Also note that roots2items is an
+  //offset array.
+  if(comm->rank() == 0){
+    dist.set_dest_ranks(Read<I32>({1, 1, 2, 2}));
+    dist.set_roots2items(LOs({0, 2, 4}));
```

Comment:
we should explain the relation of the offsets array (size and expected value of last entry) relative to the dest_ranks array

Reply:
See lines 52-56